### PR TITLE
Show "Add Date Field" for Date Picket when none of the date fields is…

### PR DESCRIPTION
**Pull request by PurwaShrivastava**
… selected instead of "Invalid Dimension"

### CODE REVIEW

File: src/qlik-date-picker.js [hunk 1]
```diff
@@ -48,21 +48,33 @@ define(["qlik", "jquery", "./lib/moment.min", "./calendar-settings", "css!./lib/
...
         }
+        function isEmpty(obj) {
+            for(var key in obj) {
+                if(obj.hasOwnProperty(key))
+                    return false;
+            }
+            return true;
+        }
         function createHtml(dateStates, DateFormat, props) {
             var html = '<div>'
```
Comment (phiols) [inline src/qlik-date-picker.js:59]: When the date picker does not have a Date Field this `<div>` tag will not be closed by corresponding `</div>`. Suggest to move line 73 to the bottom, outside of if-else statement.

File: src/qlik-date-picker.js [hunk 2]
```diff
@@ -48,21 +48,33 @@ define(["qlik", "jquery", "./lib/moment.min", "./calendar-settings", "css!./lib/
...
+                    }
+                } else {
+                html += encoder.encodeForHTML(props.defaultText);
                 }
+                html += '</span> <b class="lui-button__caret lui-caret"></b>';
+                html += '</div>';
+                html += '</div>';
             } else {
-                html += encoder.encodeForHTML(props.defaultText);
+                html += '   <i class="lui-icon lui-icon--calendar"></i>&nbsp;&nbsp;&nbsp;<span>';
```
Comment (phiols) [inline src/qlik-date-picker.js:75]: This `<span>` tag is never closed with corresponding `</span>`

File: src/qlik-date-picker.js (hunk 1 above)
Comment (PurwaShrivastava) [reply src/qlik-date-picker.js:59]: done !

File: src/qlik-date-picker.js (hunk 2 above)
Comment (PurwaShrivastava) [reply src/qlik-date-picker.js:74]: done

**Pull request merged**